make/devices.cmake

```diff
@@ -698,9 +702,160 @@ function(ADD_FPGA_TARGET)
       SOURCES ${FQDN}/${TOP}_bit.v ${TESTBENCH}
       )
   endforeach()
+
+  if(${ADD_FPGA_TARGET_EMIT_CHECK_TESTS})
+    if("${ADD_FPGA_TARGET_EQUIV_CHECK_SCRIPT}" STREQUAL "")
+      message(FATAL_ERROR "EQUIV_CHECK_SCRIPT is required if EMIT_CHECK_TESTS is set.")
+    endif()
+
+    add_check_test(
+      NAME ${NAME}_check
+      ARCH ${ARCH}
+      READ_GOLD "read_verilog ${SOURCE_FILES} $<SEMICOLON> rename ${TOP} gold"
+      READ_GATE "read_verilog ${OUT_BIT_VERILOG} $<SEMICOLON> rename ${TOP} gate"
+      EQUIV_CHECK_SCRIPT ${ADD_FPGA_TARGET_EQUIV_CHECK_SCRIPT}
+      DEPENDS ${SOURCE_FILES} ${SOURCE_FILES_DEPS} ${OUT_BIT_VERILOG}
+      )
+    add_check_test(
+      NAME ${NAME}_check_eblif
+      ARCH ${ARCH}
+      READ_GOLD "read_verilog ${SOURCE_FILES} $<SEMICOLON> rename ${TOP} gold"
+      READ_GATE "read_blif -wideports ${OUT_EBLIF} $<SEMICOLON> rename ${TOP} gate"
+      EQUIV_CHECK_SCRIPT ${ADD_FPGA_TARGET_EQUIV_CHECK_SCRIPT}
+      DEPENDS ${SOURCE_FILES} ${SOURCE_FILES_DEPS} ${OUT_EBLIF}
+      )
+  endif()
+endfunction()
+
+function(get_equiv_check_script var type)
+  # get_equiv_check_script sets var to a common EQUIV_CHECK_SCRIPT types.
+  #
+  # type can be one of the following:
+  #
+  #  * miter_and_tempinduct - Prove equivilence by creating a miter between
+  #    gold and gate, and then temporality inducting. This works on simple
+  #    cicruits.
+  #  * miter_and_tempinduct_skip_1 - Same as miter_and_tempinduct, but skips
+  #    first step.  This handles cases where the first clock is a reset step.
+  #    Outputs for the first step are undefined, but fine for the second step.
+
+  # See for more potential ideas on example EQUIV_CHECK_SCRIPT's:
+  # https://github.com/YosysHQ/yosys/issues/302#issuecomment-276019637
+  if("${type}" STREQUAL "miter_and_tempinduct")
+    set(${var}
+      "hierarchy $<SEMICOLON> proc $<SEMICOLON> clk2fflogic $<SEMICOLON> miter -equiv -flatten -ignore_gold_x -make_outputs -make_outcmp gold gate miter $<SEMICOLON> sat -prove trigger 0 -verify -tempinduct miter"
```

Comment:
Are we better maybe storing these in Yosys script files?
http://www.clifford.at/yosys/cmd_script.html

Reply:
Agreed